cross-doc-explainer.md

```diff
@@ -0,0 +1,114 @@
+the right JavaScript knobs for when the defaults are not enough.
+
+# How it works
+
+## In a nutshell
+Documents declare that they support same-origin cross-document transitions, using a
+[meta tag](#opt-in). When a navigation between two such documents takes place (and has no
+cross-origin redirects in the middle), the state of the old document is captured, in the same
+manner as same-origin transitions. When the new document is ready for the transition (i.e., when all render-blocking resources are ready), the new state is captured,
+and the transition continues as if it was a same-document transition, with the appropriate pseudo-elements.
+
+The new document can customize the style of the animation using the same techniques available for same-document transitions.
+
+Both documents can interrupt the transition or customize it using JavaScript in different phases.
+
+## Same-origin
+
+Cross-document view transitions are only enabled for same-origin navigations without a
+[cross-origin redirect](https://html.spec.whatwg.org/#unloading-documents:was-created-via-cross-origin-redirects).
+In the future we could examine exposing them in some way to same-site or cross-origin navigations.
+
+## Opt-in
+
+To enable cross-document transitions, both the old and new document need to add a meta tag. This tag
+indicates that the author wants to enable transitions for same-origin navigations to/from this
+Document.
+
+
+### Issues
+
+1. The meta tag can be used to opt-in to other navigation types going forward: same-document, same-site, etc.
+
+1. Using meta-tags prevents the declaration being controlled by media queries, which feels important for `prefers-reduced-motion`.
+See [w3c/csswg-drafts#8048](https://github.com/w3c/csswg-drafts/issues/8048).
+
+## Lifecycle
+
+![Lifecycle chart for cross-document transitions](media/mpa-chart.svg)
+
+### Capturing the old state
+
+The pold state is captured](https://drafts.csswg.org/css-view-transitions-1/#capture-old-state-algorithm) right before the old document is hidden and a new one is shown.
+In the HTML spec that moment is defined [here](https://html.spec.whatwg.org/#populating-a-session-history-entry:loading-a-document).
+This can either happen during normal navigations, when the new document is about to be created,
+in Back/Forward cache navigations, or when activating a prerendered document.
+
+Before creating the new document (or activating a cached/prerendered one), the UA would update the rendering and snapshot the old document, in the same manner a document is snapshotted for a same-document navigation.
+
+Note that currently there are no planned new events for the exit transition.
+The developer can use existing events like `navigate` or `click` to customize the
+exit transition at moments that should be late enough.
+
+#### Issues
+
+1. Should we enable skipping the transition on exit? This could be done on`pagehide`.
```

Comment:
Is `pagehide` guaranteed to be before the new Document is created? It would be awkward for the old Document to skip the transition in this event if the new Document has already accessed it via document.activeTransition?

Reply:
Yes it's guaranteed!